imx/deposit.go

```diff
@@ -37,16 +47,16 @@ type ERC721Deposit struct {
 }
 
 // NewETHDeposit instantiates a new ETHDeposit object with the given amount.
-func NewETHDeposit(amount string) *ETHDeposit {
+func NewETHDeposit(amount Wei) *ETHDeposit {
 	this := ETHDeposit{}
-	this.Amount = amount
+	this.Amount = strconv.FormatUint(amount, 10)
 	return &this
 }
 
 // NewERC20Deposit instantiates a new ERC20Deposit object with given amount and tokenAddress.
-func NewERC20Deposit(amount, tokenAddress string) *ERC20Deposit {
+func NewERC20Deposit(amount Wei, tokenAddress string) *ERC20Deposit {
```

Comment:
I think it would be a misnomer for ERC20 to use wei. Also some ERC20 tokens are not 18 decimals based.